nevergrad/optimization/optimizerlib.py

```diff
@@ -188,23 +188,38 @@ def __init__(
             budget: Optional[int] = None,
             num_workers: int = 1,
             scale: float = 1.0,
-            diagonal: bool = False
+            popsize: Optional[int] = None,
+            diagonal: bool = False,
+            fcmaes: bool = False
     ) -> None:
         super().__init__(parametrization, budget=budget, num_workers=num_workers)
         self._scale = scale
+        self._popsize = popsize
         self._diagonal = diagonal
+        self._fcmaes = fcmaes
         self._es: Optional[cma.CMAEvolutionStrategy] = None
         # delay initialization to ease implementation of variants
         self.listx: tp.List[ArrayLike] = []
         self.listy: tp.List[float] = []
         self.to_be_asked: tp.Deque[np.ndarray] = deque()
 
     @property
-    def es(self) -> cma.CMAEvolutionStrategy:
+    def es(self) -> tp.Any:  # typing not possible since cmaes not imported :(
         if self._es is None:
-            popsize = max(self.num_workers, 4 + int(3 * np.log(self.dimension)))
-            inopts = {"popsize": popsize, "randn": self._rng.randn, "CMA_diagonal": self._diagonal, "verbose": 0}
-            self._es = cma.CMAEvolutionStrategy(x0=np.zeros(self.dimension, dtype=np.float), sigma0=self._scale, inopts=inopts)
+            popsize = max(self.num_workers, 4 + int(3 * np.log(self.dimension))) if self._popsize is None else self._popsize
+            if self._fcmaes:
+                if self._diagonal:
+                    raise RuntimeError("fcmaes doesn't support diagonal=True, use fcmaes=False")
```

Comment:
This check should be performed right in the init of ParametrizedCMA, this way we'd get the error way earlier.